plugins/tileacq.py

```diff
@@ -488,20 +531,7 @@ def _get_fov(self, sd):
             raise TypeError("Unsupported object")
 
     def _cancel_acquisition(self, future):
-        """
-        Canceler of acquisition task.
-        """
-        logging.debug("Canceling acquisition...")
-
-        with future._task_lock:
-            if future._task_state == FINISHED:
-                return False
-            future._task_state = CANCELLED
-            future.running_subf.cancel()
-            logging.debug("Acquisition cancelled.")
-
-        if future._task_state == CANCELLED:
-            raise CancelledError("Acquisition cancelled")
+        return future.running_subf.cancel()
```

Comment:
This is tricky. I think that indeed in most cases this will work as well. However, if the user cancels the task just while the `running_subf` has already finished, and the next hasn't started, then the acquisition won't notice it and will keep going.

So I'd leave the original code. Maybe add a note about this case, to explain why it has to be so complicated.

Reply:
The original code didn't work, that's why I changed it! I tried to cancel the acquisition, but it just kept going.

Reply:
Oh, good catch...
Maybe it's just because it should `return True` by default. Just adding that statement at the end of the function might fix it.